diff --git a/node_modules/react-native-reanimated/RNReanimated.podspec b/node_modules/react-native-reanimated/RNReanimated.podspec
index 6cdf6ac..f12a89d 100644
--- a/node_modules/react-native-reanimated/RNReanimated.podspec
+++ b/node_modules/react-native-reanimated/RNReanimated.podspec
@@ -67,14 +67,14 @@ Pod::Spec.new do |s|
   s.dependency 'React-Core'
   s.dependency 'React-CoreModules'
   s.dependency 'React-Core/DevSupport'
-  s.dependency 'React-RCTActionSheet'
+#  s.dependency 'React-RCTActionSheet'
   s.dependency 'React-RCTNetwork'
   s.dependency 'React-RCTAnimation'
   s.dependency 'React-RCTLinking'
   s.dependency 'React-RCTBlob'
   s.dependency 'React-RCTSettings'
   s.dependency 'React-RCTText'
-  s.dependency 'React-RCTVibration'
+#  s.dependency 'React-RCTVibration'
   s.dependency 'React-RCTImage'
   s.dependency 'React-Core/RCTWebSocket'
   s.dependency 'React-cxxreact'