diff --git a/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java b/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java
index b0f62037bba60ff9a7e9fb675d8a062137bac8a8..70f893073dfe9f1e51ba5ec33303985594014cac 100644
--- a/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java
+++ b/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java
@@ -29,11 +29,12 @@ import android.view.View;
 import android.view.ViewGroup;
 import android.view.accessibility.AccessibilityNodeInfo;
 import android.widget.OverScroller;
-import android.widget.ScrollView;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.core.view.ViewCompat;
 import androidx.core.view.ViewCompat.FocusDirection;
+import androidx.core.widget.NestedScrollView;
+
 import com.facebook.common.logging.FLog;
 import com.facebook.infer.annotation.Assertions;
 import com.facebook.infer.annotation.Nullsafe;
@@ -75,7 +76,7 @@ import java.util.Set;
  * ReactHorizontalScrollView}.
  */
 @Nullsafe(Nullsafe.Mode.LOCAL)
-public class ReactScrollView extends ScrollView
+public class ReactScrollView extends NestedScrollView
     implements ReactClippingViewGroup,
         ViewGroup.OnHierarchyChangeListener,
         View.OnLayoutChangeListener,
@@ -240,7 +241,7 @@ public class ReactScrollView extends ScrollView
     if (!sTriedToGetScrollerField) {
       sTriedToGetScrollerField = true;
       try {
-        sScrollerField = ScrollView.class.getDeclaredField("mScroller");
+        sScrollerField = NestedScrollView.class.getDeclaredField("mScroller");
         sScrollerField.setAccessible(true);
       } catch (NoSuchFieldException e) {
         FLog.w(
@@ -457,7 +458,7 @@ public class ReactScrollView extends ScrollView
   }
 
   @Override
-  protected void onAttachedToWindow() {
+  public void onAttachedToWindow() {
     super.onAttachedToWindow();
     if (mRemoveClippedSubviews) {
       updateClippingRect();
