diff --git a/node_modules/react-native-gesture-handler/android/build.gradle b/node_modules/react-native-gesture-handler/android/build.gradle
index c79ff3c..8e4efc1 100644
--- a/node_modules/react-native-gesture-handler/android/build.gradle
+++ b/node_modules/react-native-gesture-handler/android/build.gradle
@@ -71,20 +71,17 @@ def reactNativeArchitectures() {
     return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
 }
 
-def libInstanceCounter = 0;
-String parsedLocation = "";
-fileTree(rootDir.parent).visit { FileVisitDetails details ->
-    def path = details.file.path
-    if (path.contains("/react-native-gesture-handler/package.json")) {
-        libInstanceCounter++;
-        parsedLocation += "- " + path.replace("/package.json", "\n")
-    }
-    if (libInstanceCounter > 1) {
+def assertNoMultipleInstallations() {
+    Set<File> files = fileTree(dir: rootDir.parent, include: "node_modules/**/react-native-gesture-handler/package.json").files
+    if (files.size() > 1) {
+        String parsedLocation = files.stream().map({ File file -> "- " + file.toString().replace("/package.json", "") }).collect().join("\n")
         String exceptionMessage = "\n[Gesture Handler] Multiple versions of Gesture Handler were detected. Only one instance of react-native-gesture-handler can be installed in a project. You need to resolve the conflict manually. Check out the documentation: https://docs.swmansion.com/react-native-gesture-handler/docs/troubleshooting#multiple-versions-of-reanimated-were-detected \n\nConflict between: \n" + parsedLocation + "\n";
         throw new Exception(exceptionMessage);
     }
 }
 
+//assertNoMultipleInstallations()
+
 def REACT_NATIVE_DIR = findNodeModulePath("react-native")
 
 repositories {
