diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java
index dee20e3..2f2c87c 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java
@@ -379,9 +379,9 @@ public class SurfaceMountingManager {
                   + actualParentId
                   + "] "
                   + " Parent: "
-                  + viewParent.getClass().getSimpleName()
+                  + viewParent
                   + " View: "
-                  + view.getClass().getSimpleName()));
+                  + view));
 
       // We've hit an error case, and `addView` will crash below
       // if we don't take evasive action (it is an error to add a View
diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/shell/MainReactPackage.kt b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/shell/MainReactPackage.kt
index 2089157..c8e1bf5 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/shell/MainReactPackage.kt
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/shell/MainReactPackage.kt
@@ -151,7 +151,7 @@ constructor(private val config: MainPackageConfig? = null) :
           ReactTextInputManager(),
           if (ReactNativeFeatureFlags.enablePreparedTextLayout()) PreparedLayoutTextViewManager()
           else ReactTextViewManager(),
-          ReactViewManager(),
+//          ReactViewManager(),
           ReactVirtualTextViewManager(),
           ReactUnimplementedViewManager())
 
@@ -191,7 +191,7 @@ constructor(private val config: MainPackageConfig? = null) :
                     PreparedLayoutTextViewManager()
                 else ReactTextViewManager()
               },
-          ReactViewManager.REACT_CLASS to ModuleSpec.viewManagerSpec { ReactViewManager() },
+//          ReactViewManager.REACT_CLASS to ModuleSpec.viewManagerSpec { ReactViewManager() },
           ReactVirtualTextViewManager.REACT_CLASS to
               ModuleSpec.viewManagerSpec { ReactVirtualTextViewManager() },
           ReactUnimplementedViewManager.REACT_CLASS to
diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewManager.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewManager.java
index 28e12d6..abc62c8 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewManager.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewManager.java
@@ -54,7 +54,7 @@ public abstract class ViewManager<T extends View, C extends ReactShadowNode>
    * null signals that View Recycling is disabled. `enableViewRecycling` must be explicitly called
    * in a concrete constructor to enable View Recycling per ViewManager.
    */
-  @Nullable private HashMap<Integer, Stack<T>> mRecyclableViews = null;
+  @Nullable public HashMap<Integer, Stack<T>> mRecyclableViews = null;
 
   public ViewManager() {
     super(null);
