diff --git a/node_modules/react-native/React/Fabric/Mounting/RCTMountingManager.mm b/node_modules/react-native/React/Fabric/Mounting/RCTMountingManager.mm
index 4f396bc..53e124b 100644
--- a/node_modules/react-native/React/Fabric/Mounting/RCTMountingManager.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/RCTMountingManager.mm
@@ -20,6 +20,8 @@
 #import <react/renderer/debug/SystraceSection.h>
 #import <react/renderer/mounting/TelemetryController.h>
 
+#import <RNScreens/RNSScreenStackComponentView.h>
+
 #import "RCTComponentViewProtocol.h"
 #import "RCTComponentViewRegistry.h"
 #import "RCTConversions.h"
@@ -47,6 +49,17 @@ static void RCTPerformMountInstructions(
 {
   SystraceSection s("RCTPerformMountInstructions");
     
+    for (auto const &mutation : mutations) {
+        if(mutation.type==ShadowViewMutation::Remove){
+            if(std::string(mutation.oldChildShadowView.componentName)=="RNSScreen"){
+                auto &parentShadowView = mutation.parentShadowView;
+                auto &parentViewDescriptor = [registry componentViewDescriptorWithTag:parentShadowView.tag];
+                auto view = (RNSScreenStackComponentView *)parentViewDescriptor.view;
+                [view screenWillGoOut];
+            }
+        }
+    }
+
   [CATransaction begin];
   [CATransaction setValue:(id)kCFBooleanTrue forKey:kCATransactionDisableActions];
   for (auto const &mutation : mutations) {
diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoHelpers.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoHelpers.java
index 6f86abb..8a91d85 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoHelpers.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoHelpers.java
@@ -21,7 +21,7 @@ public class AndroidInfoHelpers {
 
   public static final String EMULATOR_LOCALHOST = "10.0.2.2";
   public static final String GENYMOTION_LOCALHOST = "10.0.3.2";
-  public static final String DEVICE_LOCALHOST = "localhost";
+  public static final String DEVICE_LOCALHOST = EMULATOR_LOCALHOST;
 
   public static final String METRO_HOST_PROP_NAME = "metro.host";
 
diff --git a/node_modules/react-native/scripts/.packager.env b/node_modules/react-native/scripts/.packager.env
new file mode 100644
index 0000000..361f5fb
--- /dev/null
+++ b/node_modules/react-native/scripts/.packager.env
@@ -0,0 +1 @@
+export RCT_METRO_PORT=8081
diff --git a/node_modules/react-native/scripts/react_native_pods.rb b/node_modules/react-native/scripts/react_native_pods.rb
index b35ad2e..29c7967 100644
--- a/node_modules/react-native/scripts/react_native_pods.rb
+++ b/node_modules/react-native/scripts/react_native_pods.rb
@@ -384,10 +384,10 @@ def get_react_codegen_spec(options={})
     });
   end
 
-  if script_phases
-    Pod::UI.puts "[Codegen] Adding script_phases to React-Codegen."
-    spec[:'script_phases'] = script_phases
-  end
+  # if script_phases
+  #   Pod::UI.puts "[Codegen] Adding script_phases to React-Codegen."
+  #   spec[:'script_phases'] = script_phases
+  # end
 
   return spec
 end
